plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExBudget.hs

```diff
@@ -178,10 +181,33 @@ instance ExBudgetBuiltin fun () where
 
 data ExBudget = ExBudget { _exBudgetCPU :: ExCPU, _exBudgetMemory :: ExMemory }
     deriving stock (Eq, Show, Generic, Lift)
-    deriving (Semigroup, Monoid) via (GenericSemigroupMonoid ExBudget)
     deriving anyclass (PrettyBy config, NFData)
     deriving (FromJSON, ToJSON) via CustomJSON '[FieldLabelModifier (CamelToSnake)] ExBudget
 
+instance Semigroup ExBudget where
+    (ExBudget cpu1 mem1) <> (ExBudget cpu2 mem2) = ExBudget (cpu1 <> cpu2) (mem1 <> mem2)
+    stimes = scaleBudget
+
+instance Monoid ExBudget where
+    mempty = ExBudget mempty mempty
+
+-- TODO: Is there some proper numeric typeclass way of doing this?
```

Comment:
I actually deliberately didn't use `stimes`, since it somehow seemed to insist on doing a "repeated multiplication" thing instead of a raw multiplication, despite me giving an implementation that just did the multiplication. So in the end I just wrote a less-polymorphic version. I should comment that, though.

Reply:
Nope, now it's fine using stimes, gone back to that.